onnxruntime/python/tools/transformers/optimizer.py

```diff
@@ -174,7 +183,15 @@ def optimize_by_onnxruntime(
         else:
             providers.append("CUDAExecutionProvider")
 
-    onnxruntime.InferenceSession(onnx_model_path, sess_options, providers=providers, **kwargs)
+    # For large model, extract external data from model and add to session options
+    if isinstance(onnx_model, ModelProto):
+        if save_as_external_data:
+            raise ValueError("Model has external data, model path is required to load the inference session.")
```

Comment:
A better message can be like "ModelProto has external data not loaded into memory, ORT cannot create session. Please load external data before calling this function. See https://onnx.ai/onnx/repo-docs/ExternalData.html for more information".

In this way, user know the cause and solution.